### PR TITLE
Mesheryctl: Add interactive pagination for filter list

### LINKED ISSUE #10366: [mesheryctl]: Interactive Pagination for `mesheryctl filter list` command
_See [mesheryctl Command Tracker](https://bit.ly/3dqXy1q) for current status of commands._

### Current Behavior
image

### Desired Behavior
Show pagination effect similar to `mesheryctl exp model list` command.

### Mockups

---

### Contributor Resources

- ⌨️ [mesheryctl Contributing Guide](https://github.com/meshery/meshery/blob/master/mesheryctl/README.md)
- [Beginner's guide to contributing to Meshery and mesheryctl](https://youtu.be/hh_kFLZx3G4)
- [mesheryctl Command Tracker](https://docs.google.com/spreadsheets/d/1q63sIGAuCnIeDs8PeM-0BAkNj8BBgPUXhLbe1Y-318o/edit#gid=0)
- [Meshery CLI Commands and Documentation](https://docs.google.com/document/d/1xRlFpElRmybJ3WacgPKXgCSiQ2poJl3iCCV1dAalf0k/edit#heading=h.5fucij4hc5wt)
- 🙋🏾🙋🏼 Questions: [Layer5 Discussion Forum](http://discuss.meshery.io) and [Layer5 Community Slack](https://slack.meshery.io)

### PR DESCRIPTION
**Pull request by Aijeyomah**
**Notes for Reviewers**

This PR fixes #10366
- Add interactive pagination for mesheryctl filter list

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

**Comment by Aijeyomah** [comment]
@RipulHandoo Here's the PR. I left it in draft because I was unable to test, I'll need help here.

File: mesheryctl/internal/cli/root/filter/list.go [hunk 1]
```diff
@@ -92,57 +99,110 @@ mesheryctl filter list 'Test Filter' (maximum 25 filters)
...
-			UpdatedAt := fmt.Sprintf("%d-%d-%d", int(v.UpdatedAt.Month()), v.UpdatedAt.Day(), v.UpdatedAt.Year())
-			data = append(data, []string{FilterID, UserID, FilterName, CreatedAt, UpdatedAt})
-		}
-		utils.PrintToTableWithFooter([]string{"FILTER ID", "USER ID", "NAME", "CREATED", "UPDATED"}, data, []string{"Total", fmt.Sprintf("%d", response.TotalCount), "", "", ""})
-		return nil
+		return handlePagination(data, header, footer)
 	},
 }
 
+func handlePagination(data [][]string, header, footer []string) error {
```
Comment (RipulHandoo) [inline mesheryctl/internal/cli/root/filter/list.go:158]: Thank you for encapsulating this logic into a function. Please consider moving it to our `utils` module and refactoring it into a generic function. We're currently using the same logic for pagination in all `mesheryctl commands`

**Review by RipulHandoo — CHANGES_REQUESTED** [review]

**Comment by vishalvivekm** [comment]
@Aijeyomah Let's discuss this in Meshery dev call today. Add this as an agenda item to the meeting[ minutes](https://docs.google.com/document/d/1Dg4Uz9rGTuGHmYZwIt47z1j5x3LmnpxD1M1SgypAkQ0/edit?pli=1#), if you would :)

**Comment by RipulHandoo** [comment]
Some lint check failure. @Aijeyomah

File: mesheryctl/internal/cli/root/system/model.go [hunk 2]
```diff
@@ -349,7 +302,7 @@ mesheryctl exp model search [query-text]
 	Args: func(_ *cobra.Command, args []string) error {
 		const errMsg = "Usage: mesheryctl exp model search [query-text]\nRun 'mesheryctl exp model search --help' to see detailed help message"
 		if len(args) == 0 {
-			return fmt.Errorf("Search term is missing\n\n%v", errMsg)
+			return fmt.Errorf("search term is missing\n\n%v", errMsg)
```
Comment (RipulHandoo) [inline mesheryctl/internal/cli/root/system/model.go:305]:
```suggestion
			return utils.ErrInvalidArgument(errors.New("Please provide a model name. " + errMsg))
```

**Review by RipulHandoo — DISMISSED** [review]

**Review by RipulHandoo — APPROVED** [review]

File: mesheryctl/internal/cli/root/connections/list.go [hunk 3]
```diff
@@ -119,48 +118,11 @@ mesheryctl exp connections list --page 2
...
-
-				if event.Key == keyboard.KeyEnter || event.Key == keyboard.KeyArrowDown {
-					startIndex += maxRowsPerPage
-					endIndex = min(len(rows), startIndex+maxRowsPerPage)
-				}
-
-				if startIndex >= len(rows) {
-					break
-				}
+			maxRowsPerPage := 25
```
Comment (MUzairS15) [inline mesheryctl/internal/cli/root/connections/list.go:121]: There's a global default page size variable already defined, please use that.

File: mesheryctl/internal/cli/root/filter/list.go [hunk 4]
```diff
@@ -92,53 +98,68 @@ mesheryctl filter list 'Test Filter' (maximum 25 filters)
 					UpdatedAt := fmt.Sprintf("%d-%d-%d %d:%d:%d", int(v.UpdatedAt.Month()), v.UpdatedAt.Day(), v.UpdatedAt.Year(), v.UpdatedAt.Hour(), v.UpdatedAt.Minute(), v.UpdatedAt.Second())
 					data = append(data, []string{FilterID, FilterName, CreatedAt, UpdatedAt})
 				}
-				utils.PrintToTableWithFooter([]string{"FILTER ID", "NAME", "CREATED", "UPDATED"}, data, []string{"Total", fmt.Sprintf("%d", response.TotalCount), "", ""})
-				return nil
+				header = []string{"FILTER ID", "NAME", "CREATED", "UPDATED"}
```
Comment (MUzairS15) [inline mesheryctl/internal/cli/root/filter/list.go:101]: Ideally, we want to rely on the `JSON` tags defined with the struct and move away from this hardcoded list.

@Aijeyomah Something to take on as the next action item?

**Review by MUzairS15 — APPROVED** [review]
Perfect @Aijeyomah great work on extracting the repetitive logic into a common util. 🚀

File: mesheryctl/internal/cli/root/filter/list.go (hunk 4 above)
Comment (Aijeyomah) [reply mesheryctl/internal/cli/root/filter/list.go:101]: > Ideally, we want to rely on the `JSON` tags defined with the struct and move away from this hardcoded list.
> 
> 
> 
> @Aijeyomah Something to take on as the next action item?

Yeah, I can

File: mesheryctl/internal/cli/root/connections/list.go (hunk 3 above)
Comment (MUzairS15) [reply mesheryctl/internal/cli/root/connections/list.go:121]: // @Aijeyomah

File: mesheryctl/internal/cli/root/filter/list.go (hunk 4 above)
Comment (MUzairS15) [reply mesheryctl/internal/cli/root/filter/list.go:101]: Great will you open an issue to track this and start working (if you want to)?

**Pull request merged**